src/inference/src/file_utils.cpp

```diff
@@ -2,6 +2,9 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#ifndef FILE_UTILS_CPP
+#define FILE_UTILS_CPP
```

Comment:
Header guard for .cpp file looks strange and would not work as expected.

Reply:
Header guards look very strange. And won't work if we include this file in several object libraries. Please provide another way to avoid the issue.